proxy/src/main/java/com/velocitypowered/proxy/connection/util/ServerListPingHandler.java

```diff
@@ -44,11 +47,29 @@ public ServerListPingHandler(VelocityServer server) {
 
   private ServerPing constructLocalPing(ProtocolVersion version) {
     VelocityConfiguration configuration = server.getConfiguration();
+    List<ServerPing.SamplePlayer> samplePlayers;
+    if (configuration.getSamplePlayersInPing()) {
+      List<ServerPing.SamplePlayer> unshuffledPlayers = server.getAllPlayers().stream()
+              .map(p -> {
+                if (p.getPlayerSettings().isClientListingAllowed()) {
+                  return new ServerPing.SamplePlayer(p.getUsername(), p.getUniqueId());
+                } else {
+                  return new ServerPing.SamplePlayer(
```

Comment:
We can cache this object in a `static final`, `ServerPing.SamplePlayer` is an immutable class.

Reply:
Do you mean like holding a list of sample players statically as a variable in a class (maybe `VelocityServer`)? If that's the case then maybe I should change the current shuffling behavior so it either only shuffles once per join/leave event, or I could just have it insert each player join at a random point in the list.

What do you think?

Reply:
Sorry I wasn't clear enough, meant that we should cache the "unknown player" instance, not the entire list, i.e. there's no need to create a new "unknown player" `SamplePlayer` instance every time `p.getPlayerSettings().isClientListingAllowed()` returns `false`.

Reply:
Is this what you meant? ac484c972f4e4c3e7312f629b31fb5e7d6f369ce